src/include/miopen/tensor_reorder_util.hpp

```diff
@@ -0,0 +1,286 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2020-2022 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef MIOPEN_TENSOR_REORDER_UTIL_HPP_
+#define MIOPEN_TENSOR_REORDER_UTIL_HPP_
+
+#include <miopen/batched_transpose_sol.hpp>
+#include <miopen/general_tensor_reorder_sol.hpp>
+#include <miopen/miopen.h>
+#include <miopen/kernel_info.hpp>
+#include <miopen/op_kernel_args.hpp>
+#include <miopen/execution_context.hpp>
+#include <vector>
+
+namespace miopen {
+struct TensorReorderSolution
+{
+
+    virtual ~TensorReorderSolution()                      = default;
+    virtual solver::KernelInfo GetKernel() const          = 0;
+    virtual std::vector<OpKernelArg> GetKernelArg() const = 0;
+    virtual std::string GetKernelName() const             = 0;
+    virtual bool IsSkippable() const                      = 0;
+    virtual size_t GetSize() const                        = 0;
```

Comment:
`IsSkippable()` used in HOST side to check the special cases that either tensor height or width equal = 1. In such cases, we don't need to conduct batched transpose operation, since the transposed tensor layout has exactly same memory layout as before.
`GetSize()` used in HOST side to get tensor total memory size used in `handle.CreateSubBuffer().`
You can find both usage example in `/src/solver/conv_asm_implicit_gemm_gtc_wrw_nhwc.cpp`.

Reply:
@aska-0096 Good! Then can you please do the following:
- Document the purpose of these functions to the abstract base class
- Use these functions in the test
- Rename GetSize to GetOutputTensorSize. Why:
  - We are creating sub-buffers for output tensors, right?
  - In this specific transformation, sizes of input and output tensors are equal, but other transformations may have them different.
  - If/when we need GetInputTensorSize, then it's easy to add it.

Thanks!

Reply:
:green_circle: Resolved at acceptable level in #1481